### PR TITLE
SAML: add <mdui:UIInfo> element examples

### PR DESCRIPTION
**Pull request by localguru**
add some <mdui:UIInfo> element examples for saml2_config in homeserver.yaml

### CODE REVIEW

File: changelog.d/8718.misc [hunk 1]
```diff
@@ -0,0 +1 @@
+add some mdui:UIInfo element examples for saml2_config in homeserver.yaml.
```
Comment (anoadragon453) [inline changelog.d/8718.misc:1]:
```suggestion
Add some `mdui:UIInfo` element examples for `saml2_config` in the homeserver config.
```

File: synapse/config/saml2_config.py [hunk 2]
```diff
@@ -271,6 +271,24 @@ def generate_config_section(self, config_dir_path, server_name, **kwargs):
...
+            #  information_url:
+            #    - lang: en
+            #      text: "https://example.com/terms-of-service"
+            #  privacy_statement_url:
+            #    - lang: en
+            #      text: "https://example.com/privacy-policy"
+            #  logo:
+            #    text: "https://example.com/logo.svg"
+            #    width: "200"
+            #    height: "80"
```
Comment (anoadragon453) [inline synapse/config/saml2_config.py:290]: According to https://github.com/IdentityPython/pysaml2/blob/59d6fa5df06989525d2d7e5b8762bbfa3485ab42/src/saml2/metadata.py#L268-L276 it looks as though `logo` can also be a list of dicts, which looks more flexible. Any reason that's not show in the example?

**Review by anoadragon453 — COMMENTED** [review]
I struggled to find any documentation of this option in [pysaml2's docs](https://pysaml2.readthedocs.io/en/latest/howto/config.html).

However I did find how it is processed in [pysaml2's source code](https://github.com/IdentityPython/pysaml2/blob/59d6fa5df06989525d2d7e5b8762bbfa3485ab42/src/saml2/metadata.py#L227-L278) and it looks valid!

Just a tiny question below, otherwise lgtm.

File: synapse/config/saml2_config.py (hunk 2 above)
Comment (localguru) [reply synapse/config/saml2_config.py:290]: @anoadragon453 no reasons, feel free to modify it. This is basically the config I use in my SAML config.

Comment (localguru) [reply synapse/config/saml2_config.py:290]: @anoadragon453 like for all elements multiple languages are possible like

 ```
ui_info:
      display_name:
        - lang: en
          text: "Display Name is the descriptive name of your service."
       - lang: de
          text: "Display Name ist der beschreibende Name Ihres Dienstes"
```

So I think for `logo` it has to look like this
``` 
           logo:
               - lang: en
                 text: "https://example.com/logo_en.svg"
                 width: "200"
                 height: "80"
               - lang: de
                 text: "https://example.com/logo_de.svg"
                 width: "200"
                 height: "80"
```

Comment (localguru) [reply synapse/config/saml2_config.py]: @anoadragon453 Andrew, I checked list of dicts for logo element with my IdP an this generates valid metadata `/_matrix/saml2/metadata.xml`:

saml2_config:
```
ui_info:
           [...]
           logo:
              - lang: en
                text: "https://example.com/logo_en.svg"
                width: "200"
                height: "80"
              - lang: de
                text: "https://example.com/logo_de.svg"
                width: "200"
                height: "80"
```

/saml2/metadata.xml`:
```
<ns1:Logo height="86" width="200" xml:lang="de">
https://example.com/logo_de.svg
</ns1:Logo>
<ns1:Logo height="86" width="200" xml:lang="en">
https://example.com/logo_en.svg
</ns1:Logo>
```
The central Shibboleth IdP at my university also evaluates the language for logos. If I switch the browser between DE and EN, the corresponding logo is also displayed on the registration form of the IdP.

Furthermore I have added the `keywords` element.

**Comment by clokep** [comment]
I'm unsure about documenting all the options in here, should we more clearly point to pysaml2 docs? I also don't want people to have to figure that library out though so... probably ok to add it.

**Comment by localguru** [comment]
I understand. The problem with the ui_info options is, that these are not documented. It took me several hours to find this out. I think these options are important, especially because of `privacy_statement_url` and `information_url` can refer to privacy statements and terms of use. This is mandatory for the IdP in the German Research Network. Options such as `organization` and `contact_person` would then also have to be discussed, as these are not mandatory for the pure functionality of SAML registration.

File: synapse/config/saml2_config.py (hunk 2 above)
Comment (anoadragon453) [reply synapse/config/saml2_config.py]: Glad to hear, thank you for testing!

**Review by anoadragon453 — APPROVED** [review]
I'm inclined to agree with @localguru on the issue of verbosity, mainly on the importance of certain options and their lack of documentation in where we link to (perhaps a good candidate for an upstream PR from somebody as well :)

lgtm

**Pull request merged**